docs/quickstart.rst

```diff
@@ -0,0 +1,553 @@
+Once ``handle_auth_success`` is called, it collects the identities linked to the user using the ``get_identity_from_auth`` we just mentioned. 
+Once identities are successfully collected, the method registered via ``'@multipass.identity_handler'`` decorator is called. A method with this decorator must be implemented in your application. Check the :ref:`identity_handler` part of this documentation for more info.
+
+Failed authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In case the authentication was unsuccessful, and ``MultipassException`` was raised, ``handle_auth_error`` flashes the ``MULTIPASS_FAILURE_MESSAGE`` and if ``redirect_to_login`` argument is set, it redirects to ``MULTIPASS_LOGIN_ENDPOINT``
+
+
+Local provider logout
+~~~~~~~~~~~~~~~~
+
+Process of logging out local providers is the same as with external providers.
+Please check the :ref:`external_logout` part of this guide.
+
+.. _identities:
+
+Identities
+----------
+
+To retrieve ``IdentityInfo`` object  your  ``IdentityProvider`` must implement ``get_identity`` method. Example from ``StaticIdentityProvider``:
+
+.. code-block:: python
+
+    def get_identity(self, identifier):
+        user = self.settings['identities'].get(identifier)
+        if user is None:
+            return None
+        return IdentityInfo(self, identifier, **user)
+
+Same applies for searching identities. There you accept ``criteria`` dictionary as a filter for your search. Example from ``StaticIdentityProvider``:
+
+.. code-block:: python
+
+    def search_identities(self, criteria, exact=False):
+        for identifier, user in iteritems(self.settings['identities']):
+            for key, values in iteritems(criteria):
+                user_value = user.get(key)
+                user_values = set(user_value) if isinstance(user_value, (tuple, list)) else {user_value}
+                if not any(user_values):
+                    break
+                elif exact and not user_values & set(values):
+                    break
+                elif not exact and not any(sv in uv for sv, uv in itertools.product(values, user_values)):
+                    break
+            else:
+                yield IdentityInfo(self, identifier, **user)
+
+Once implemented on your  ``IdentityProvider`` you can also use method  ``search_identities`` on ``Multipass`` object which will search among all the providers and yield all the ``identity_info``  matching the criteria specified in the argument. 
+
+.. code-block:: python
+
+	criteria['name'] = 'Guinea Pig'
+	criteria['email'] = 'guinea.pig@example.com'
+	results = list(multipass.search_identities(exact=False, **criteria))
+
+.. _groups:
+
+Groups
+-------
+
+Providers can divide users into groups. This is usually based on access rights and competences of users. For example group of admins, content managers, regular users etc. These groups should be specified in the configuration settings of identity provider. Example from our ``'test_identity_provider'``: 
```

Comment:
"[...] on **the** access rights and competences of users, for example: whether they are admins, content managers, regular users, etc."
"configuration settings of **the** identity provider"